src/hb-mutex.hh

```diff
@@ -81,20 +81,26 @@ typedef int hb_mutex_impl_t;
 
 #else
 
-#error "Could not find any system to define mutex macros."
-#error "Check hb-mutex.hh for possible resolutions."
+#include <mutex>
+#include <new>
+typedef std::mutex              hb_mutex_impl_t;
+#define hb_mutex_impl_init(M)   HB_STMT_START { new (M) hb_mutex_impl_t; } HB_STMT_END
+#define hb_mutex_impl_lock(M)   (*(M)).lock ()
+#define hb_mutex_impl_unlock(M) (*(M)).unlock ()
+#define hb_mutex_impl_finish(M) HB_STMT_START { (M)->~hb_mutex_impl_t(); } HB_STMT_END
 
 #endif
 
 
 struct hb_mutex_t
 {
-  hb_mutex_impl_t m;
+  // Create space for, but do not initialize m
+  alignas(hb_mutex_impl_t) char m[sizeof(hb_mutex_impl_t)];
 
-  void init   () { hb_mutex_impl_init   (&m); }
-  void lock   () { hb_mutex_impl_lock   (&m); }
-  void unlock () { hb_mutex_impl_unlock (&m); }
-  void fini ()   { hb_mutex_impl_finish (&m); }
+  void init   () { hb_mutex_impl_init   ((hb_mutex_impl_t*)&m); }
```

Comment:
Aren't the `&` redundant here?

Reply:
I suppose arrays do like to decay to pointers. Note that this code is casting `char[N]*`. The proposal to remove the `&` (not taking the the address) would cast `char*` instead. It makes no difference here since we're just casing the pointer either way, but it would make a difference if doing pointer arithmetic.